build/common.sh

```diff
@@ -833,3 +833,43 @@ function kube::release::gcs::publish_latest() {
   echo "+++ gsutil cat ${latest_file_dst}:"
   gsutil cat ${latest_file_dst}
 }
+
+# Publish a new latest.txt, but only if the release we're dealing with
+# is newer than the contents in GCS.
+function kube::release::gcs::publish_latest_official() {
+  local new_version=${KUBE_GCS_LATEST_CONTENTS}
+  local latest_file_dst="gs://${KUBE_GCS_RELEASE_BUCKET}/${KUBE_GCS_LATEST_FILE}"
+
+  local version_regex="^v(0|[1-9][0-9]*)\\.(0|[1-9][0-9]*)\\.(0|[1-9][0-9]*)$"
+  [[ ${new_version} =~ ${version_regex} ]] || {
+    echo "!!! publish_latest_official passed bogus value: '${new_version}'" >&2
+    exit 1
+  }
+
+  local version_major="${BASH_REMATCH[1]}"
+  local version_minor="${BASH_REMATCH[2]}"
+  local version_patch="${BASH_REMATCH[3]}"
+
+  local gcs_version
+  gcs_version=$(gsutil cat ${latest_file_dst})
+
+  [[ ${gcs_version} =~ $version_regex ]] || {
+    echo "!!! ${latest_file_dst} contains invalid release version, can't compare: '${gcs_version}'" >&2
+    exit 1
+  }
+
+  local gcs_version_major="${BASH_REMATCH[1]}"
+  local gcs_version_minor="${BASH_REMATCH[2]}"
+  local gcs_version_patch="${BASH_REMATCH[3]}"
+
+  if [[ "${gcs_version_major}" -gt "${version_major}" ]] ||
```

Comment:
Hmmm this one is tricky because your check will think that 1.9.1 > 3.3.3 because 9 > 3 in the middle...

What you want is to check for `-gt`, in which case you're done, otherwise check for `-lt`, in which case it's not greater and you're also done, and then if doesn't fall on either (assuming `-eq` is the only other possibility) then proceed to check the next element of the version string.

Also, not sure we have v0.5 or v0.6 anywhere, without the last .0... Not sure if you actually care about that anyways, given modern versions seem to always include the full triplet.